src/main/scala/org/constellation/session/SessionTokenService.scala

```diff
@@ -0,0 +1,86 @@
+package org.constellation.session
+
+import cats.effect.Concurrent
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import io.chrisdavenport.fuuid.FUUID
+import io.circe.generic.semiauto._
+import io.circe.{Decoder, Encoder}
+import org.constellation.session.SessionTokenService._
+import org.http4s.util.CaseInsensitiveString
+
+class SessionTokenService[F[_]: Concurrent] {
+
+  private[session] val peersTokens: Ref[F, Map[String, Token]] = Ref.unsafe(Map.empty)
+
+  private[session] val ownToken: Ref[F, Option[Token]] = Ref.unsafe(None)
+
+  def createAndSetNewOwnToken: F[Token] =
+    for {
+      token <- createToken
+      _ <- ownToken.modify(_ => (token.some, ()))
+    } yield token
+
+  def getOwnToken: F[Option[Token]] =
+    ownToken.get
+
+  def updatePeerToken(ip: String, token: Token): F[Unit] =
+    peersTokens.modify(tokens => (tokens + (ip -> token), ()))
+
+  def getPeerToken(ip: String): F[Option[Token]] =
+    peersTokens.get.map(_.get(ip))
+
+  def verifyPeerToken(ip: String, oHeaderToken: Option[Token]): F[TokenVerificationResult] =
+    for {
+      oPeerToken <- getPeerToken(ip)
+      verificationResult = SessionTokenService.compareTokens(oPeerToken, oHeaderToken)
+    } yield verificationResult
+
+  def clearOwnToken: F[Unit] =
+    ownToken.modify(_ => (None, ()))
+
+  def clearPeerToken(ip: String): F[Unit] =
+    peersTokens.modify(tokens => (tokens - ip, ()))
+
+  def clear: F[Unit] =
+    for {
+      _ <- clearOwnToken
+      _ <- peersTokens.modify(_ => (Map.empty, ()))
+    } yield ()
+
+  private[session] def createToken: F[Token] =
+    for {
+      uuid <- FUUID.randomFUUID
+      token = new Token(uuid.toString)
+    } yield token
+
+}
+
+object SessionTokenService {
+  class Token(val value: String) extends AnyVal
+
+  object Token {
+    implicit val tokenEncoder: Encoder[Token] = deriveEncoder
+    implicit val tokenDecoder: Decoder[Token] = deriveDecoder
+  }
+
+  val XSessionToken = CaseInsensitiveString("X-Session-Token")
```

Comment:
I personally think `X-Session-Token` will look better but it's up to you :D

Reply:
Yeah I think I agree :D

Reply:
But unfortunately "-" is not allowed in variable name :(

Reply:
````val `X-Session-Token` = CaseInsensitiveString("X-Session-Token")```